agent/agent.go

```diff
@@ -343,6 +343,28 @@ func (a *Agent) processNextTask(ctx context.Context, nt *apimodels.NextTaskRespo
 	if nt.ShouldTeardownGroup {
 		// Tear down the task group if the task group is finished.
 		a.runTeardownGroupCommands(ctx, tc)
+
+		// Terminate host if a task groups is complete in a single host distro.
+		if a.opts.SingleTaskDistro {
+			err := a.comm.DisableHost(ctx, a.opts.HostID, apimodels.DisableInfo{Reason: "Single task distro ran a task"})
```

Comment:
So is the api calls like: single task host starts, picks up a task group, finishes it, hits this DisableHost endpoint (which I assume tells Evergreen not to run a task on this host again?), continues the loop cycle (which Evergreen will not send a new task), the agent exists, and the host will get decommissioned/removed?

If this fails, is it possible for the single task host to pick up another unrelated task? Could we lock the host from picking up more tasks but when we dispatch the task to the host (i.e. what I think this API call is doing but when the task is dispatched) so we know for sure that host won't run another task?

Reply:
Ah got it! That makes sense to me!